pkg/controller/upgrades.go

```diff
@@ -406,3 +409,14 @@ func (c *Controller) upgrade500(ctx context.Context, tenant *miniov2.Tenant) (*m
 	}
 	return c.updateTenantSyncVersion(ctx, tenant, version500)
 }
+
+// Upgrades the sync version to v5.0.12
+// in this version we save most of the minio env variables in the config secret, except for the env variables needed
+// for k8s runtime
+func (c *Controller) upgrade5012(ctx context.Context, tenant *miniov2.Tenant) (*miniov2.Tenant, error) {
+	err := c.saveTenantConfiguration(ctx, tenant)
```

Comment:
Even if we do this once on upgrade the users can easily overwrite it by re-applying their local yamls, better to move this to the sidecar